src/capture-screenshot.js

```diff
@@ -0,0 +1,145 @@
+const puppeteer = require('puppeteer');
+const {performance} = require('perf_hooks');
+
+const devicePixelRatio = 1.0;
+const maxTimeInSec = 30;
+
+const timeDelta = (start, end) => {
+  return ((end - start) / 1000).toPrecision(3);
+}
+
+const htmlTemplate = (options) => {
+  const {width, height, inputPath, libPort, backgroundColor} = options;
+  return `
+    <html>
+      <head>
+        <meta name="viewport" content="width=device-width, initial-scale="${devicePixelRatio}">
+        <script type="module"
+          src="http://localhost:${libPort}/model-viewer.js">
+        </script>
+        <style>
+          body {
+            margin: 0;
+          }
+          model-viewer {
+            --progress-bar-color: transparent;
+            width: ${width};
+            height: ${height};
+          }
+        </style>
+      </head>
+      <body>
+        <model-viewer
+          style="background-color: ${backgroundColor};"
+          background-color=""
+          id="snapshot-viewer"
+          interaction-prompt="none"
+          src="${inputPath}" />
+      </body>
+    </html>
+  `
+}
+
+const captureScreenshot = async function (options) {
```

Comment:
Most of this code is tweaked and borrowed from how <model-viewer> does their fidelity testing here: https://github.com/google/model-viewer/blob/master/packages/render-fidelity-tools/src/artifact-creator.ts#L253. I could have just taken the relevant `poster-visibility`hack they implemented into our existing flow but to be honest this was much cleaner with less timeout logic.